metrics/resource_view_test.go

```diff
@@ -147,6 +152,50 @@ func (m metricExtract) String() string {
 	return fmt.Sprintf("%s:%d", m.Key(), m.Value)
 }
 
+func initSdFake(sdFake *stackDriverFake) error {
+	if err := sdFake.start(); err != nil {
+		return err
+	}
+	conn, err := grpc.Dial(sdFake.address, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	newStackdriverExporterFunc = func(o stackdriver.Options) (view.Exporter, error) {
+		o.MonitoringClientOptions = append(o.MonitoringClientOptions, option.WithGRPCConn(conn))
+		return newOpencensusSDExporter(o)
+	}
+	// File: must exist, be json of credentialsFile, and type must be a jwtConfig or oauth2Config
+	tmp, err := ioutil.TempFile("", "metrics-sd-test")
+	if err != nil {
+		return err
+	}
+	credentialsContent := []byte(`{"type": "service_account"}`)
+	if _, err := tmp.Write(credentialsContent); err != nil {
+		return err
+	}
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", tmp.Name())
+	return nil
+}
+
+func sortMetrics() cmp.Option {
```

Comment:
You can make this a `var` so you don't need to call the function, but I'm fine either way.